tests/AgDatabaseMove.Unit/BackupOrder.cs

```diff
@@ -44,51 +60,158 @@ private static List<BackupMetadata> ListBackups()
     {
-      var agDatabase = new Mock<IAgDatabase>();
-      agDatabase.Setup(agd => agd.RecentBackups()).Returns(_listBackups);
-      var backupChain = new BackupChain(agDatabase.Object);
+      var list = GetBackupList();
+      list.RemoveAll(b => b.BackupType == BackupFileTools.BackupType.Log);
+      return list;
+    }
+    
+    private static List<BackupMetadata> GetBackupListWithoutDiff()
+    {
+      var list = GetBackupList();
+      list.RemoveAll(b => b.BackupType == BackupFileTools.BackupType.Diff);
+      return list;
+    }
+
+    private static List<BackupMetadata> GetBackupListWithStripes()
+    {
+      var list = GetBackupList();
+      var listWithStripes = CloneBackupMetaDataList(list).ToList();
+      listWithStripes.ForEach(b => {
+        var path = b.PhysicalDeviceName.Split('.');
+        b.PhysicalDeviceName = $"{path[0]}_striped.{path[1]}";
+      });
+      list.AddRange(listWithStripes);
+      return list;
+    }
+
+    private static List<BackupMetadata> GetBackupListWithStripesAndDuplicates()
+    {
+      var listWithStripes = GetBackupListWithStripes();
+      var duplicate = CloneBackupMetaDataList(listWithStripes);
+      listWithStripes.AddRange(duplicate);
+      return listWithStripes;
+    }
+
+    private static List<BackupMetadata> GetBackupListWithoutFull()
+    {
+      var list = GetBackupList();
+      list.RemoveAll(b => b.BackupType == BackupFileTools.BackupType.Full);
+      return list;
+    }
+
+    private static void VerifyListIsAValidBackupChain(IEnumerable<BackupMetadata> backupChain)
+    {
+      var backupChainList = backupChain.ToList();
 
-      var expected = _listBackups.OrderBy(bu => bu.FirstLsn);
+      var listOfFileNames = backupChainList.Select(b => b.PhysicalDeviceName).ToList();
+      Assert.True(listOfFileNames.Count == listOfFileNames.Distinct().Count());
+
+      var orderedBackups = CloneBackupMetaDataList(backupChainList)
+        .OrderBy(b => b.LastLsn)
+        // 'PhysicalDeviceName' ordering doesn't matter but need it here to simplify this check for LSN ordering
+        .ThenBy(b => b.PhysicalDeviceName).ToList();
+      Assert.True(backupChainList.SequenceEqual(orderedBackups, new BackupMetadataEqualityComparer()));
+
+      // first one has to be full
+      var full = backupChainList.FirstOrDefault();
+      Assert.NotNull(full);
+      Assert.True(full.BackupType == BackupFileTools.BackupType.Full);
+
+      // removing all the striped files of the full backup
+      backupChainList.RemoveAll(b => b.BackupType == BackupFileTools.BackupType.Full &&
+                                     b.CheckpointLsn == full.CheckpointLsn);
+
+      var otherFullBackups = backupChainList.RemoveAll(b => b.BackupType == BackupFileTools.BackupType.Full);
+      Assert.Equal(0, otherFullBackups);
+
+      // next one can be diff or log
+      var currentBackup = backupChainList.FirstOrDefault();
+      if(currentBackup == null) {
+        return;
+      }

```

Comment:
I get that this works and all.... but it seems like you could reduce this 50 line function into a 5 line do()while() loop checking that lastLsn == next.FirstLsn for the entire chain

Reply:
yes, so that is pretty much what I have done to check the log backups:
https://github.com/factset/AgDatabaseMove/blob/8abb6abb67b5a0ce5a3c69239d1dcb86df910c5d/tests/AgDatabaseMove.Unit/BackupOrder.cs#L171-L179

But all of this before is to make sure:
- The first backup in the chain is a full backup
- There are no other full backups (excluding the stripes of the first one)
- If there is a diff backup, then it is the only one and that its backupLsn == full.checkpointLsn
- And this will also make sure that the order is maintained full then diff (if any) then logs (if any) 

But I do see your point, looking at this now maybe there is a way to refactor this to be more simple
Let me give it a try

Reply:
Refactored it to be a bit shorter - let me know what you think